### DISCUSSION TITLE
Trouble Acquiring camera frame

### DISCUSSION THREAD

**Discussion by SoppierBrute9** [bonsai-rx/bonsai#1519] [Q&A]
I am working with Bonsai to synchronize camera capture from two cameras, a FLIR black fly s and a Basler ace. When I use a camera capture and video writer I get a runtime error that says unable to acquire camera frame. Using the tutorial for multiple cameras I still get the same runtime error.

**Comment by bruno-f-cruz** [comment]
I don't remember those cameras having generic drivers, so I doubt CameraCapture will work. You will probably have to use their specific packages (eg Bonsai.Spinnaker).

**Reply by SoppierBrute9** [reply]
Using the specific packages I get a build error "no method overload found for the given argument"

**Comment by bruno-f-cruz** [comment]
Please attach the workflow. Thanks.

On Tue, 15 Aug 2023 at 10:26, SoppierBrute9 ***@***.***>
wrote:

> Using the specific packages I get a build error "no method overload found
> for the given argument"
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/bonsai-rx/discussions/1519#discussioncomment-6732748>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVZBB5NPHA3MZKHNXJ2PCTXVOWKXANCNFSM6AAAAAA3RK5GY4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Sent from my phone. Apologies for being brief.

**Reply by SoppierBrute9** [reply]
![IMG_2622](https://github.com/bonsai-rx/bonsai/assets/142339490/d4c3f11d-5255-487c-b7ac-02e009002228)

**Comment by bruno-f-cruz** [comment]
The output of those nodes is a structure that has an image and
corresponding metadata. If you want to concatenate the frames you will have
to extract the image portion by right clicking on the node. Also, the size
of the frames must be compatible.

Additionally, and as has been discussed several times in the past, the
strategy you are using will NOT  synchronize the cameras at the hardware
level. There will likely be some jitter in the two streams which might lead
to a few frames of misalignment (not sure this is critical for you
application).


On Tue, 15 Aug 2023 at 10:32, SoppierBrute9 ***@***.***>
wrote:

> [image: IMG_2622]
> <https://user-images.githubusercontent.com/142339490/260789621-720dd4f4-27ce-460f-ab6f-7e7529cf0d2b.jpeg>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/bonsai-rx/discussions/1519#discussioncomment-6732781>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVZBB4SXHZC7QNRXFNLPMTXVOXCNANCNFSM6AAAAAA3RK5GY4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Sent from my phone. Apologies for being brief.

**Reply by SoppierBrute9** [reply]
A few frames off is not a big deal, what I am trying to do is have the cameras start and stop recording when I start and stop the workflow. I am not able to right click on the Spinnaker capture node.

**Reply by bruno-f-cruz** [reply]
Can you provide a bit more information? What error do you get when you try to right-click on the operator?

**Reply by SoppierBrute9** [reply]
It says unhandled exception has occurred in your application...A procedure imported by 'SpinnakerNET_v140.dll' could not be loaded

**Comment by bruno-f-cruz** [comment]
Have you installed the spinnaker camera software as instructed in the
package description?

On Tue, 15 Aug 2023 at 10:47, SoppierBrute9 ***@***.***>
wrote:

> It says unhandled exception has occurred in your application...A procedure
> imported by 'SpinnakerNET_v140.dll' could not be loaded
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/bonsai-rx/discussions/1519#discussioncomment-6732903>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVZBBZ2KGYWRIH2JHQ2ADLXVOY4XANCNFSM6AAAAAA3RK5GY4>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Sent from my phone. Apologies for being brief.

**Reply by SoppierBrute9** [reply]
I uninstalled the spinnaker package and reinstalled it, im not sure if I need to do anything more. I am still having the issue.

**Reply by bruno-f-cruz** [reply]
What software and version have you installed?

**Reply by SoppierBrute9** [reply]
I have spinview 3.1.0.79 and I am using the most recent Spinnaker package.
Even when I just do pyloncapture to videowriter I get no method overload.

**Reply by SoppierBrute9** [reply]
![IMG_2623](https://github.com/bonsai-rx/bonsai/assets/142339490/ce2e895e-3649-4df2-b518-1f074a210809)
This is the error I get when just using the pyloncapture for the basler camera

**Reply by bruno-f-cruz** [reply]
Please use the package version instructed in the package description.

**Reply by SoppierBrute9** [reply]
I downloaded all the software instructed in the packages and the error I get is the input arrays must have the same shape except in the dimension corresponding to the axis. How do I fix this?
